compiler/passes/convert-uast.cpp

```diff
@@ -1129,7 +1267,7 @@ struct Converter {
     if (!node->isExpressionLevel()) return false;
     if (node->iterand()->isZip()) return false;
     if (node->numStmts() != 1) return false;
-    if (node->index()) return false;
+    if (node->index() && node->stmt(0)->isConditional()) return false;
```

Comment:
Is this something like `[i in 1..100]`? How could that be an array type?

Reply:
It may not be an array type, but the old parser considers it one today in some cases. The problem is the AST printer uses `maybeArrayType` to print out square braces. So if we change that we can change the logic here.

Reply:
OK, in that case please just add a comment saying that same thing